support-frontend/assets/pages/contributions-landing/contributionsLandingActions.js

```diff
@@ -89,13 +93,14 @@ const updateContributionType = (contributionType: ContributionType): ((Function)
     dispatch(setFormSubmissionDependentValue(() => ({ type: 'UPDATE_CONTRIBUTION_TYPE', contributionType })));
   };
 
-const updatePaymentMethod = (paymentMethod: PaymentMethod): Action => {
-  // PayPal one-off redirects away from the site before hitting the thank you page
-  // so we need to store the payment method in the storage so that it is available on the
-  // thank you page in all scenarios.
-  storage.setSession('selectedPaymentMethod', paymentMethod);
-  return ({ type: 'UPDATE_PAYMENT_METHOD', paymentMethod });
-};
+const updatePaymentMethod = (paymentMethod: PaymentMethod): Action =>
+  (dispatch: Function): void => {
+    // PayPal one-off redirects away from the site before hitting the thank you page
+    // so we need to store the payment method in the storage so that it is available on the
+    // thank you page in all scenarios.
+    storage.setSession('selectedPaymentMethod', paymentMethod);
+    dispatch(setFormSubmissionDependentValue(() => ({ type: 'UPDATE_PAYMENT_METHOD', paymentMethod })));
```

Comment:
Changing the payment method is now a 'Form Submission Dependent Value' because the 'Contribute' button needs to be disabled only if the Stripe form is incomplete

Reply:
I'm confused about why we need to disable the button completely instead of just error handling like we do in other cases of required fields not being filled in

Reply:
@jesseyuen thoughts?